### ISSUE TITLE
Make better use of horizontal space for chat

### ISSUE THREAD

**Issue by Hawdjer** [microsoft/vscode-copilot-release#60]
Instead of opening the chat in the Extension bar directly, give an option to open it as a window that I can drag/split in the editor being able to see the Solution Explorer at the same time.

**Comment by arivero** [comment]
Also it could be interesting if the chat were aware of itself as a tool inside visual code, giving help about how to do this, or if it is impossible to do. As of today, it seems to belibe it can be dragged elsewhere.

**Comment by jpenalbae** [comment]
Its already implemented:
> Ctrl + Shift + P -> chat: open Editor (github copilot)

You can also make quick questions without opening a new tab or the sidebar. This is especially useful when used with a keybinding:
> Ctrl + Shift + P -> chat: ask quick question

**Comment by rogerio-barreto** [comment]
@jpenalbae that's good to hear, although after open it up in the window seems the content is limited to max 600px;

Can we get it fluid/100% as we have in the current side bars?

![image](https://github.com/microsoft/vscode-copilot-release/assets/85616763/91bfbefc-b816-4ed4-93a7-3715a1defde9)

**Comment by roblourens** [comment]
Yeah, this is supported with chat editors. How wide do you want it to be though? I think that reading very wide columns of text is uncomfortable, and you generally won't have that much content on a single line in a chat.

**Comment by RogerBarreto** [comment]
@roblourens Since we as developers can resize the window at our own desire, would make more sense to occupy the whole window `flex` and not be restricted.

**Comment by roblourens** [comment]
Same reason github lays out issue text in narrow columns- people generally don't like reading a single line of text across the full monitor.

I'm measuring chatGPT's at about 650 px. Mine is 600. Github is about 800. I can go wider but I don't think I'm totally out of line here.

**Comment by RogerBarreto** [comment]
I completely agree that for a Pure no coding chat text experience this is fine and make totally sense but for developers, well you will end up in a very common scenario where you add extra UX pain to use the horizontal bar in Code Snipets most of the times forcing us to copy what is in there to somewhere else to see the whole picture.

Like the example above.

**Comment by roblourens** [comment]
That's actually a good point, maybe I can keep text wrapped in columns for readability for make use of more width for code blocks. Thanks

**Comment by daviddossett** [comment]
It's an interesting idea. It does create other problems. It could hurt readability if your eye has to jump around from the text to the code. It also would be quite silly to have such a large width in cases where it isn't needed. 

<img width="1408" alt="CleanShot 2023-06-07 at 15 02 27@2x" src="https://github.com/microsoft/vscode-copilot-release/assets/25163139/03f23b32-c7a3-4198-ae0c-46764e027876">

**Comment by roblourens** [comment]
I think I could size it to the content so it would only grow like that when the content is wide, if that helps

**Comment by TylerLeonhardt** [comment]
This kinda reminds me of `Toggle Centered Layout` command which centers your editor and gives you the ability to widen or shrink the window.

**Comment by roblourens** [comment]
Now you get 850px

**Comment by RogerBarreto** [comment]
Waaay better, thanks

**Comment by lifeiteng** [comment]
> Now you get 850px

Can we set the width > 850px after the vscode be installed?

**Comment by DaveWelling** [comment]
This problem is more pronounced with the inline chat:
![image](https://github.com/microsoft/vscode-copilot-release/assets/1242456/b5bcf4dc-511b-49f4-8d8d-288d13293da0)
I'm able to slightly widen the code by dragging an (invisible) divider between the red and green areas:
![image](https://github.com/microsoft/vscode-copilot-release/assets/1242456/c869b8b5-3774-46fc-bcd3-6c373f31ec11)
But I have no affordance to see the rest of the code.  There is no horizontal scrollbar and I cannot widen the dialog box itself. To actually read this code, I must either:

1. Accept the code
2. Copy and paste
3. Drag a highlighted selection back and forth across the code so that the box scrolls to show what I am selecting.

**Comment by roblourens** [comment]
That would be a separate issue @DaveWelling, feel free to open a new issue. Seems like it should actually be inserting the code in this scenario and not rendering a text edit in the list.

**Comment by qiulang** [comment]
@roblourens hi,

I am a bit confused with the status of issue. My chat window is so small at bottom right of vscode (as https://code.visualstudio.com/docs/copilot/getting-started-chat shows), can I make the chat window bigger or full screen?


Thanks

**Comment by roblourens** [comment]
You can resize the sidebar by dragging or open chat in an editor by running "Open Chat in Editor"

**Comment by qiulang** [comment]
Yes, I know that but what I mean was that the chatbox (that blue square) is too short and I can make it bigger ?

![open chat in editor](https://github.com/user-attachments/assets/d7eb39ed-86b7-4268-8aad-d8d2f8b86539)

BTW, Chatgpt indeed also has a small chatbox, compared to claude.

**Comment by roblourens** [comment]
No, but if you type multiple lines of text, then it will expand automatically up to a point.

**Comment by arpadtegda** [comment]
"people generally don't like reading a single line of text across the full monitor." - you can always scale DOWN your window in that case, but if I want to maximize it to full window width, the text column width won't follow my expectations. :-(
